internal/command/jsonplan/values.go

```diff
@@ -168,12 +171,12 @@ func marshalPlannedValues(changes *plans.Changes, schemas *tofu.Schemas) (Module
 }
 
 // marshalPlanResources
-func marshalPlanResources(changes *plans.Changes, ris []addrs.AbsResourceInstance, schemas *tofu.Schemas) ([]Resource, error) {
+func marshalPlanResources(changeMap map[string]*plans.ResourceInstanceChangeSrc, ris []addrs.AbsResourceInstance, schemas *tofu.Schemas) ([]Resource, error) {
 	var ret []Resource
 
 	for _, ri := range ris {
-		r := changes.ResourceInstance(ri)
-		if r.Action == plans.Delete {
+		r, ok := changeMap[ri.String()]
+		if !ok || r.Action == plans.Delete {
```

Comment:
nit: `changes.ResourceInstance` returns nil if there is nothing found. Then, it would panic on `r.Action == plans.Delete` check. In the current implementation, we would skip the resource if it is not found, assuming this is a `plans.Delete` action. However, it might silence some critical dev error (because panic is hidden). I would prefer to keep the original behavior here.

Reply:
Good point! Done.